### DISCUSSION TITLE
Can't sign ethereum transaction in kotlin

### DISCUSSION THREAD

**Discussion by tajition** [trustwallet/wallet-core#1717] [Q&A] [answered]
**Describe the bug**
Cannot sign transaction on ethereum. I have imported hd wallet and created the wallets. I can get their addresses and fetch the balance using external API. But I cant sign a transaction. 

I have two unresolved references:
1) toHexByteArray() 
2) Ethereum.Transaction

here are my imports 
```
import wallet.core.jni.PrivateKey
import wallet.core.jni.CoinType.ETHEREUM
import wallet.core.jni.proto.Ethereum
import wallet.core.jni.proto.Ethereum.SigningOutput
import com.trustwallet.core.app.utils.toHexByteArray        // Error here. unresolved reference app
import wallet.core.jni.proto.Ethereum.TransactionMode    // Error here. unresolved reference TransactionMode
import com.trustwallet.core.app.utils.Numeric                   // Error here. unresolved reference app 
```
I tried deleting all imports and relying on android studio to auto import but it doesn't find them. 

**To Reproduce**
1) Import TrustWalletCore
2) Create wallets
3) Try to sign eth transaction

I am using this example to sign a transaction: [https://github.com/trustwallet/wallet-core/blob/master/android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/ethereum/TestEthereumTransactionSigner.kt](TestEthereumTransactionSigner.kt)

I have copied the testEthereumTransactionSigning verbatim but removed the assertions instead opting to just get the hex string like this
`val hexString = Numeric.toHexString(output.encoded.toByteArray())`

**Expected behavior**
Get transaction hex string 

**Screenshots**
not sure these help but.
![image](https://user-images.githubusercontent.com/9453588/138011340-c3dbc845-f529-4469-b93d-4607cd9136ac.png)
![image](https://user-images.githubusercontent.com/9453588/138011376-88102665-81d8-42c2-b9a4-fe440bde43a0.png)

**Additional context**
Fairly new to kotlin and android development. A pointer in the right direction is much appreciated. 

Edit: 
I have this import in my build.gradle depedencies 

```
dependencies {
    implementation "com.trustwallet:wallet-core:2.6.3"
    implementation "com.squareup.okhttp3:okhttp:4.9.0"
    implementation "com.google.code.gson:gson:2.8.8"
    implementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.13.0"
    implementation "com.google.zxing:core:3.4.0"
    implementation "com.budiyev.android:code-scanner:2.1.0"
    
    implementation "androidx.core:core-ktx:1.6.0"
    implementation "androidx.appcompat:appcompat:1.3.1"
    implementation "com.google.android.material:material:1.4.0"
    implementation "androidx.constraintlayout:constraintlayout:2.1.1"
    implementation "androidx.navigation:navigation-fragment-ktx:2.3.5"
    implementation "androidx.navigation:navigation-ui-ktx:2.3.5"
    implementation "androidx.legacy:legacy-support-v4:1.0.0"
    testImplementation "junit:junit:4.+"
    androidTestImplementation "androidx.test.ext:junit:1.1.3"
    androidTestImplementation "androidx.test.espresso:espresso-core:3.4.0"
}
```

not sure if something I imported might be conflicting as well

**Comment by optout21** [comment]
There are many many Kotlin .kt test files, please consult those.
I think the missing imports are:
`import com.trustwallet.core.app.utils.toHexByteArray`
`import wallet.core.jni.proto.Ethereum`

**Reply by tajition** [reply]
I have the imports but they are unresolved 

```
import com.trustwallet.core.app.utils.toHexByteArray
import com.trustwallet.core.app.utils.Numeric
import wallet.core.jni.proto.Ethereum.TransactionMode
```

the following imports are unresolved

![image](https://user-images.githubusercontent.com/9453588/138078237-ee1ff100-6e39-4c6a-b669-dd9f882ba51c.png)

Edit 1:
I pushed to code github and pulled it on my laptop.  The weird things is 
`import wallet.core.jni.proto.Ethereum.TransactionMode`
Still doesnt work but I can replace it with 
`import wallet.core.jni.proto.Ethereum.Transaction`
and it will the unresolved error will go away for this import. 
But back on my desktop the error persists either way

Edit 2:
Think the app folder is missing altogether somehow in my import.
![image](https://user-images.githubusercontent.com/9453588/138080737-ed3be58f-39dc-47ed-8908-00db1e2f79ea.png)

Edit 3:
I fixed the 
`import wallet.core.jni.proto.Ethereum.TransactionMode`
by replacing it with 
`import wallet.core.jni.proto.Ethereum.Transaction` 
and rerunning the gradle build. 
Still can't get com.trustwallet.core.app to resolved though

**Comment by tajition** [comment, marked as answer]
Solved the issue by using this example from sample app

https://github.com/trustwallet/wallet-core/blob/2.6.30/samples/android/app/src/main/java/com/trust/walletcore/example/MainActivity.kt

Tried examples following the test files but those didn't work without utils class. Maybe they will work if I take the required methods from the sample app and add them directly.
